index.js

```diff
@@ -441,6 +442,33 @@ Router.prototype.route = function route (path) {
   return route
 }
 
+/**
+ * List all registered routes.
+ *
+ * @return {Array} An array of route paths
+ * @public
+ */
+Router.prototype.mapRoutes = function mapRoutes () {
+  const routes = []
+  const stack = this.stack
+
+  const options = {
+    strict: this.strict,
+    caseSensitive: this.caseSensitive
+  }
+
+  const routeMap = new Map()
+
+  collectRoutes(stack, '', routeMap, options)
```

Comment:
I understand why this is being done this way, but I am always concerned when using a function with side effects like this. I think we should consider having this public router method take an optional input for the `Map` and then instead of recursing with `collectRoutes` just use `router.routes()` (assuming we go with that naming I suggested above).